core/src/main/scala/org/apache/spark/errors/SparkCoreErrors.scala

```diff
@@ -31,6 +31,25 @@ import org.apache.spark.storage.{BlockId, BlockManagerId, BlockNotFoundException
  * Object for grouping error messages from (most) exceptions thrown during query execution.
  */
 object SparkCoreErrors {
+  def unexpectedPy4JServerError(other: Object): Throwable = {
+    new RuntimeException(s"Unexpected Py4J server ${other.getClass}")
+  }
+
+  def eofExceptionWhileReadPortNumberError(
+      daemonModule: String,
+      daemonExitValue: Option[Int] = null): Throwable = {
+    var msg = s"EOFException occurred while reading the port number from $daemonModule's" +
+      s" stdout"
+    if (daemonExitValue != null) {
```

Comment:
Should be `daemonExitValue.isDefined`. Could be reformulated as
`val msg = s" ... stdout" + daemonExitValue.map(v => s" and terminated with code: $v.").getOrElse("")`